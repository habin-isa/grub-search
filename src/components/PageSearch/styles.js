import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background: #ced6ff;
  width: 100%;
  height: 100vh;
  align-items: center;
`;

export const SearchContainer = styled.div`
  display: flex;
  flex-direction: column;
  color: #333;
  width: 100%;
  padding: 50px;
`;

export const Title = styled.div`
  font-size: 35px;
`;

export const Subtitle = styled.div`
  font-size: 15px;
  padding-top: 40px;
  line-height: 25px;
`;

export const Link = styled.a`
  text-decoration: none;
  color: #2a4afc;
  font-weight: bold;
  &:hover {
    color: white;
  }
`;

export const Icon = styled.img`
  width: 30%;
`;

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const SearchWrapper = styled.form`
  display: flex;
  flex-direction: column;
  padding-top: 100px;
  font-size: 22px;
`;

export const Label = styled.label`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Input = styled.input`
  background: white;
  border: none;
  padding: 10px;
  margin-left: 10px;
  color: #333;
  font-size: 15px;
  width: 40%;
  margin-top: 10px;
`;

export const SubmitContainer = styled.div`
  text-align: right;
  margin-top: 10px;
`;

export const Submit = styled.input`
  background: #fa9629;
  color: white;
  font-size: 15px;
  padding: 10px 50px;
  width: fit-content;
  &:hover {
    background: #fcac2a;
    cursor: pointer;
  }
`;
