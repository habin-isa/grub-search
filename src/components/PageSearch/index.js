import React from 'react';
import * as S from './styles';
import FoodIcon from './assets/spaghetti.png';
import { array, func, obj } from 'prop-types';

const PageSearch = ({ handleSearchSubmit, handleSearchChange, userInput }) => {
  return (
    <S.Wrapper>
      <S.SearchContainer>
        <S.Title>Search for a restaurant or a category for your grub</S.Title>
        <S.Subtitle>
          To access the FourSquare API, create a developer account and obtain client credentials (ID and secret), from{' '}
          <S.Link href="https://developer.foursquare.com/docs/places-api/" target="_blank" rel="noopener noreferrer">
            here.
          </S.Link>
          <br />
          Out of ideas? Why not search for <i>Grill</i> or <i>Coffee!</i>
        </S.Subtitle>
        <S.SearchWrapper onSubmit={handleSearchSubmit}>
          <S.Label>
            Foursquare API Client ID:
            <S.Input type="text" name="clientId" onChange={handleSearchChange} value={userInput.clientId} />
          </S.Label>
          <S.Label>
            Foursqaure API Client Secret:
            <S.Input type="text" name="clientSecret" onChange={handleSearchChange} value={userInput.clientSecret} />
          </S.Label>
          <S.Label>
            Restaurant name:
            <S.Input type="text" name="name" onChange={handleSearchChange} value={userInput.name} />
          </S.Label>
          <S.SubmitContainer>
            <S.Submit type="submit" value="Submit" />
          </S.SubmitContainer>
        </S.SearchWrapper>
      </S.SearchContainer>
      <S.IconContainer>
        <S.Icon src={FoodIcon} alt="food-icon" />
      </S.IconContainer>
    </S.Wrapper>
  );
};

PageSearch.propTypes = {
  handleSearchSubmit: func,
  handleSearchChange: func,
  userInput: obj
};

PageSearch.defaultProps = {
  handleSearchSubmit: () => {},
  handleSearchChange: () => {},
  userInput: {}
};

export default PageSearch;
