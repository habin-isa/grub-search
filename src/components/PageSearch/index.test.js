import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import PageSearch from '.';

describe('PageSearch component', () => {
  const props = {
    handleSearchSubmit: () => {},
    handleSearchChange: () => {},
    userInput: {
      name: 'Kim',
      clientId: '123213',
      clientSecret: '123213'
    },
    inputError: 0
  };
  it('should render', () => {
    const { container } = render(<PageSearch {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with correct userInput data', () => {
    shallow(<PageSearch userInput={props.userInput} />);
  });
});
