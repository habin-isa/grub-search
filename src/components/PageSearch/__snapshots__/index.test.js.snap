// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSearch component should render 1`] = `
<div
  class="sc-AxjAm cAfKVR"
>
  <div
    class="sc-AxirZ fTDDfI"
  >
    <div
      class="sc-AxiKw evJipK"
    >
      Search for a restaurant or a category for your grub
    </div>
    <div
      class="sc-AxhCb dlLDwE"
    >
      To access the FourSquare API, create a developer account and obtain client credentials (ID and secret), from
       
      <a
        class="sc-AxhUy iDIzqx"
        href="https://developer.foursquare.com/docs/places-api/"
        rel="noopener noreferrer"
        target="_blank"
      >
        here.
      </a>
      <br />
      Out of ideas? Why not search for 
      <i>
        Grill
      </i>
       or 
      <i>
        Coffee!
      </i>
    </div>
    <form
      class="sc-Axmtr hZhPkX"
    >
      
      <label
        class="sc-AxmLO fEFzgM"
      >
        Foursquare API Client ID:
        <input
          class="sc-fzozJi fewFLE"
          name="clientId"
          type="text"
          value="123213"
        />
      </label>
      <label
        class="sc-AxmLO fEFzgM"
      >
        Foursqaure API Client Secret:
        <input
          class="sc-fzozJi fewFLE"
          name="clientSecret"
          type="text"
          value="123213"
        />
      </label>
      <label
        class="sc-AxmLO fEFzgM"
      >
        Restaurant name:
        <input
          class="sc-fzozJi fewFLE"
          name="name"
          type="text"
          value="Kim"
        />
      </label>
      <div
        class="sc-fzoLsD lnvcMU"
      >
        <input
          class="sc-fzpans ckUvuc"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
  <div
    class="sc-AxheI iulBeS"
  >
    <img
      alt="food-icon"
      class="sc-AxgMl iZoIAo"
      src="spaghetti.png"
    />
  </div>
</div>
`;
